e2e/2.x/basic/components/ExtendedTsConfig.vue

```diff
@@ -0,0 +1,34 @@
+<template>
+  <div>
+    {{ exclamationMarks }}
+    <type-script-child />
+  </div>
+</template>
+
+<script lang="ts">
+import TypeScriptChild from './TypeScriptChild.vue'
+
+import moduleRequiringEsModuleInterop from './ModuleRequiringEsModuleInterop'
+
+// The default import above relies on esModuleInterop being set to true in order to use it from
+// an import statement instead of require. This option is configured in the tsconfig.base.json,
+// so if we are no longer fully processing the tsconfig options (extended from a base config)
+// this test should fail. This was one of the only reliable ways I could get a test to fail if
+// these conditions are not being met and happen to be the use-case which was triggering errors
+// in my config setup.
```

Comment:
In my project, we were setting esModuleInterop in a shared base config, and `Vue.extend(...)` was throwing a TypeError when running the test because it "could not read property 'extend' from `undefined`".